src/runtime/arcs-types/particle-spec.ts

```diff
@@ -381,6 +381,18 @@ export class ParticleSpec {
     return [...this.slotConnections.keys()];
   }
 
+  /**
+   * Returns true if there's a consume or provide connection named @name.
+   */
+  hasSlotConnectionName(name: string): boolean {
+    if (this.slotConnections.has(name)) {
+      return true;
+    }
+    return [...this.slotConnections.values()].map(connection => {
+      return connection.provideSlotConnections.map(pc => pc.name).includes(name);
+    }).reduce((a, b) => a || b);
```

Comment:
@shans will the node+chrome optimisers actually be able to convert this to an early exit? I'm not seriously considering the performance of this, but it does seem like the kind of situation where the underlying impl should just be a look up on a map from name to slot.